RecoHI/HiJetAlgos/plugins/HiFJGridEmptyAreaCalculator.cc

```diff
@@ -0,0 +1,471 @@
+  band_(iConfig.getParameter<double>("bandWidth")),
+  hiBinCut_(iConfig.getParameter<int>("hiBinCut")),
+  doCentrality_(iConfig.getParameter<bool>("doCentrality")),
+  keepGridInfo_(iConfig.getParameter<bool>("keepGridInfo"))
+{
+  pfCandsToken_ = consumes<reco::PFCandidateCollection>(iConfig.getParameter<edm::InputTag>( "pfCandSource" ));
+  mapEtaToken_ = consumes<std::vector<double> >(iConfig.getParameter<edm::InputTag>( "mapEtaEdges" ));
+  mapRhoToken_ = consumes<std::vector<double> >(iConfig.getParameter<edm::InputTag>( "mapToRho" ));
+  mapRhoMToken_ = consumes<std::vector<double> >(iConfig.getParameter<edm::InputTag>( "mapToRhoM" ));
+  jetsToken_ = consumes<edm::View<reco::Jet> >(iConfig.getParameter<edm::InputTag>( "jetSource" ));
+  centralityBinToken_ = consumes<int>(iConfig.getParameter<edm::InputTag>("CentralityBinSrc"));
+
+  //register your products
+  produces<std::vector<double > >("mapEmptyCorrFac"); 
+  produces<std::vector<double > >("mapToRhoCorr");
+  produces<std::vector<double > >("mapToRhoMCorr");
+  produces<std::vector<double > >("mapToRhoCorr1Bin");
+  produces<std::vector<double > >("mapToRhoMCorr1Bin");
+  //rho calculation on a grid using median
+  if(keepGridInfo_){
+    produces<std::vector<double > >("mapRhoVsEtaGrid");
+    produces<std::vector<double > >("mapMeanRhoVsEtaGrid");
+    produces<std::vector<double > >("mapEtaMaxGrid");
+    produces<std::vector<double > >("mapEtaMinGrid");
+  }
+}
+
+
+HiFJGridEmptyAreaCalculator::~HiFJGridEmptyAreaCalculator()
+{
+ 
+   // do anything here that needs to be done at desctruction time
+   // (e.g. close files, deallocate resources etc.)
+
+}
+
+// ------------ method called to produce the data  ------------
+void
+HiFJGridEmptyAreaCalculator::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  using namespace edm;
+  //Values from rho calculator  
+  edm::Handle<std::vector<double>> mapEtaRanges;
+  iEvent.getByToken(mapEtaToken_, mapEtaRanges);
+  
+  edm::Handle<std::vector<double>> mapRho;
+  iEvent.getByToken(mapRhoToken_, mapRho);
+
+  edm::Handle<std::vector<double>> mapRhoM;  
+  iEvent.getByToken(mapRhoMToken_, mapRhoM);
+
+  // if doCentrality is set to true calculate empty area correction
+  // for only events with hiBin > hiBinCut  
+  int hiBin = -1;
+  bool doEmptyArea = true; 
+  if(doCentrality_){
+    edm::Handle<int> cbin;
+    iEvent.getByToken(centralityBinToken_,cbin);
+    hiBin = *cbin;
+   
+    if(hiBin < hiBinCut_) doEmptyArea = false;
+  }
+  
+  //Define output vectors
+  int neta = (int)mapEtaRanges->size();
+   
+  std::auto_ptr<std::vector<double>> mapToRhoCorrOut ( new std::vector<double>(neta-1,1e-6));
+  std::auto_ptr<std::vector<double>> mapToRhoMCorrOut ( new std::vector<double>(neta-1,1e-6));
+  std::auto_ptr<std::vector<double>> mapToRhoCorr1BinOut ( new std::vector<double>(neta-1,1e-6));
+  std::auto_ptr<std::vector<double>> mapToRhoMCorr1BinOut ( new std::vector<double>(neta-1,1e-6));
```

Comment:
When I do the suggested change I get this compilation error:
    put(std::auto_ptr<PROD> product, std::string const& productInstanceName);
     ^~~
/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc600/cms/cmssw/CMSSW_8_1_X_2016-07-19-1100/src/FWCore/Framework/interface/Event.h:130:5: note:   template argument deduction/substitution failed:
/data/dgulhan/JetRECOPR/recentPR/CMSSW_8_1_X_2016-07-19-1100/src/RecoHI/HiJetAlgos/plugins/HiFJGridEmptyAreaCalculator.cc:169:48: note:   'std::shared_ptr<std::vector<double> >' is not derived from 'std::auto_ptr<_Up>'
     iEvent.put(mapEtaMaxGridOut,"mapEtaMaxGrid");